File: app/models/ImdbApiMovie.java
package models;

import play.Logger;

import com.google.gson.annotations.SerializedName;

public class ImdbApiMovie{
	
	@SerializedName("imdbID")
	public String id;
	
	@SerializedName("Title")
	public String title;
	
	@SerializedName("imdbRating")
	public String rating;
	
	@SerializedName("imdbVotes")
	public String votes;
	
	@SerializedName("Runtime")
	public String runtime;
	
	@SerializedName("Poster")
	public String poster;
	
	@SerializedName("Genre")
	public String genre;
	
	@SerializedName("Plot")
	public String plot;
	
	@SerializedName("Year")
	public Integer year;
	
	@SerializedName("Released")
	public String released;
	
	@SerializedName("Director")
	public String director;
	
	public int getPercentRating(){
		if(rating == null){
			return -1;
		}
		try{
			return (int)(Double.parseDouble(rating) * 10);
		} catch (NumberFormatException e) {
			Logger.warn(e.getMessage());
		}
		return -1;
	}
	
	public int getDecimalRating(){
		return getPercentRating() / 10;
	}
	
	public String getUrl(){
		return "http://www.imdb.com/title/" + id;
	}
	
	@Override
	public String toString() {
		return id + " " + title;
	}

}

File: app/models/helper/ImdbApiMovie.scala
package models.helper

import play.api.Logger

import scala.util.Try

case class ImdbApiMovie(
  imdbID: String,
	Title: String,
	imdbRating: Option[String],
	imdbVotes: String,
	Runtime: String,
	Poster: String,
	Genre: String,
	Plot: String,
	Year: String,
	Released: String,
	Director: String
){

  val id = imdbID
  val title = Title
  val rating = imdbRating
  val votes = imdbVotes
  val runtime = Runtime
  val poster = Poster
  val genre = Genre
  val plot = Plot
  val year = Try(Year.takeWhile(_ != '-').toInt).getOrElse(0) // year can be in format 2013 or 2013–2015
  val released = Released
  val director = Director
	
  lazy val getPercentRating = {
    rating.flatMap{ rat =>
      Try((rat.toDouble * 10).toInt).toOption
    }
	}
	
	lazy val getDecimalRating = {
		getPercentRating.map(_ / 10)
	}
	
	lazy val getUrl = "http://www.imdb.com/title/" + id
}